### PR TITLE
Adjust Http tests for UAP behavior differences

### LINKED ISSUE #22159: SendAsync_Cancel_CancellationTokenPropagates throw wrong exception on UAP
HttpClientHandler test  SendAsync_Cancel_CancellationTokenPropagates() is throwing OperationCanceledException on UAP.  But it throws TaskCanceledException on .NET Core.

### LINKED ISSUE #22163: GetAsync_StatusCode99_ExpectedException fails on UAP
Running the test with a statuscode value of '99' fails the test on UAP.  On UAP, the GetAsync() does not throw an exception. On .NET Core, it throws.

>ERROR:    System.Net.Http.Functional.Tests.HttpClientHandlerTest.GetAsync_StatusCodeOutOfRange_ExpectedException(statusCode: 99) [FAIL]
 Assert.Throws() Failure
 Expected: typeof(System.Net.Http.HttpRequestException)
 Actual:   (No exception was thrown)
 
Need to investigate to know which behavior is correct.

### LINKED ISSUE #22192: SendAsync_SendRequestUsingNoBodyMethodToEchoServerWithContent_NoBodySent fails in UAP
The HttpClientHandler test fails with using the TRACE verb.

>ERROR:    System.Net.Http.Functional.Tests.HttpClientHandlerTest.SendAsync_SendRequestUsingNoBodyMethodToEchoServerWithContent_NoBodySent(method: "TRACE", secureServer: False) [FAIL]
 System.Net.Http.HttpRequestException : An error occurred while sending the request.
 ---- System.Runtime.InteropServices.COMException : The text associated with this error code could not be found.
        The requested operation is invalid

Needs investigation. Most likely a WinRT HttpClient behavior difference.

### PR DESCRIPTION
**Pull request by davidsh**
Made changes to some UAP tests due to known platform differences.
Differences will be documented in the AppCompat docs.

Opened some new issues for bugs to investigate.

Fixes #22159
Fixes #22163
Fixes #22192

### CODE REVIEW

**Comment by davidsh** [comment]
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot Test Outerloop UWP NETNative x86 Release Build
@dotnet-bot Test Outerloop Windows x64 Debug Build

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 1]
```diff
@@ -1201,31 +1222,18 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
...
-
         [OuterLoop] // TODO: Issue #11345
         [Theory]
+        [InlineData(99)]
         [InlineData(1000)]
         public async Task GetAsync_StatusCodeOutOfRange_ExpectedException(int statusCode)
         {
+            if (PlatformDetection.IsUap && statusCode == 99)
+            {
+                // UAP platform allows this status code due to historical reasons.
```
Comment (CIPop) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Can we fix this in CoreFX code? (still throw even if the underlying HTTP stack didn't error?)

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 2]
```diff
@@ -1675,6 +1692,17 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
                 }
             }
 
+            if (PlatformDetection.IsUap)
```
Comment (CIPop) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Nit: here and above: wouldn't it be more readable if it has a single `if (IsUap && method=="TRACE")` ?

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Will fix a new issue to discuss.

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 2 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: ok...will fix in next PR with other changes.

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: #22196

**Pull request merged**